include/cutlass/gemm/collective/xe_mma.hpp

```diff
@@ -210,31 +234,31 @@ struct CollectiveMma<
     (void)residue_mnk;
     (void)thread_idx;
     (void)smem_buf;
-
-    auto tiled_copy_a = make_xe_2d_copy(atom_load_A{}.with(mainloop.mA),
-                                             Layout<Shape<_1, Int<SubgroupSize>>>{});
-    auto tiled_copy_b = make_xe_2d_copy(atom_load_B{}.with(mainloop.mB),
-                                             Layout<Shape<_1, Int<SubgroupSize>>>{});
-
-    // Partition the copying of A and B tiles across the threads
-    auto thr_copy_A = tiled_copy_a.get_slice(thread_idx);
-    auto thr_copy_B = tiled_copy_b.get_slice(thread_idx);
+    
+    auto thr_copy_A2 = mainloop.copy_A.get_slice(thread_idx);
+    auto thr_copy_B2 = mainloop.copy_B.get_slice(thread_idx);
 
     // Instantiate the MMA object and get thread slice
     TiledMma tiled_mma;
-    auto thr_mma = tiled_mma.get_slice(thread_idx);
+    // To make all threads in a warp have the same global tensors pass in the index of thread 0 in each warp
+    auto thr_mma = tiled_mma.get_slice(thread_idx & ~15);
 
-    // Partition fragment
-    Tensor fragment_A = thr_mma.partition_fragment_A(gA(_, _, 0));
-    Tensor fragment_B = thr_mma.partition_fragment_B(gB(_, _, 0));
+    // Partition
+    Tensor tCgA = thr_mma.partition_A(gA);
+    Tensor tCgB = thr_mma.partition_B(gB);
 
-    // Retile for copy
-    Tensor copy_tCrA = thr_copy_A.retile_D(fragment_A);
-    Tensor copy_tCrB = thr_copy_B.retile_D(fragment_B);
+    // Create fragments
+    // TODO(Codeplay): fix this, this is probably not general
+    Tensor tCrA = make_tensor<ElementA>(tCgA(_,_,_,0).shape());
+    Tensor tCrB = make_tensor<ElementB>(tCgB(_,_,_,0).shape(), make_stride(_1{}, shape<0>(tCgB) * shape<2>(tCgB), shape<0>(tCgB)));
```

Comment:
This too line does not seems to match what you are aiming to do

Reply:
what do you mean with that?